### ISSUE TITLE
YOLOv2

### ISSUE THREAD

**Issue by iwasakishuto** [iwasakishuto/Papers-With-Comments#4]
# YOLOv2
- Better, Faster, Stronger
- Submitted on Sun, 25 Dec 2016 (v1)

### Link
- [paper](https://pjreddie.com/media/files/papers/YOLO9000.pdf)
- [Project Webpage](https://pjreddie.com/darknet/yolo/)

### Summary
Introduce **YOLOv2** and **YOLO9000**

- YOLOv2 is SOTA (on standard detection tasks like **PASCAL, VOC, and COCO**) and faster detection systems. It **can be run at a variety of image sizes.**
- YOLO9000 is a real-time framework for detection **more than 9000 object categories** by **jointly optimizing detection (COCO dataset) and classification (ImageNet) simultaneously.**

#### Improvement
[YOLO](https://github.com/iwasakishuto/Papers-With-Comments/issues/2) suffers from a variety of **shortcomings** relative to SOTA detection systems.

- Makes a significant number of **localization errors.**
- Has relatively low **recall** compared to region proposal-based methods. (ex.R-CNN)

Thus, they focus mainly on **improving recall and localization while maintaining classification accuracy and processing speed. (Not scaling up network.)**

They tried many ideas and a summary of results are as follows.

|     #         |YOLO  |  |  |  |  |  |  | YOLOv2  |    
|-:|:-:|:-:|:-:|:-:|:-:|:-:|:-:|:-:|
|batch norm?         |  |X |X |X |X |X |X |X |X
|hi-res classifier?  |  |  |X |X |X |X |X |X |X
|convolutional?      |  |  |  |X |X |X |X |X |X
|anchor boxes?       |  |  |  |X |X
|new network?        |  |  |  |  |X |X |X |X |X
|dimension priors?   |  |  |  |  |  |X |X |X |X
|location prediction?|  |  |  |  |  |X |X |X |X
|passthrough?        |  |  |  |  |  |  |X |X |X
|multi-scale?        |  |  |  |  |  |  |  |X |X
|hi-res detector?    |  |  |  |  |  |  |  |  |X
|VOC2007 mAP |63.4 |65.8 |69.5 |69.2 |69.6 |74.4 |75.4 |76.8 |78.6|

- **Batch Normalization**
- **High Resolution Classifier**: pretrain the network at the higher resolution (224×224→448×448) to adjust its filter to work better on higher resolution input.
- **Convolutional With Anchor Boxes**: Remove the fully connected layer and use **anchor boxes** to predict bounding boxes. This
    - simplifies the problem.
    - makes it easier for the network to learn.
    - enables models to predict more than a thousand bounding boxes.
- **Dimension Clusters**: Run k-means clustering on the training set bounding boxes to automatically find good dimensions for **anchor boxes.**
- **Direct location prediction**: Instead of predicting offset they follow the approach of YOLO and predict location coordinates relative to the location of the frid cell.
- **Fine-Grained Features**: Expand the feature maps so that **it has access to fine frained features.**
- **Multi-Scale Training**: Since YOLOv2 only uses convolutional and pooling layers it **can be resized on the fly.**

#### Future Works
- They try to use **"image segmentation"** tasks to improve YOLO.
- Looking for ways to **bring different sources and structures of data together** to make stronger models of the visual world.
